src/tape/ensemble.py

```diff
@@ -1587,3 +1587,326 @@ def sf2(self, sf_method="basic", argument_container=None, use_map=True):
             result = self.batch(calc_sf2, use_map=use_map, argument_container=argument_container)
 
             return result
+
+
+"""
+    The following package-level methods can be used to create a new Ensemble object 
+    by reading in the given data source.
+"""
+
+
+def read_pandas_dataframe(
```

Comment:
"read" doesn't seem like the right word, since you've already read the originating file, and you're just constructing the ensemble from already-in-memory pandas. `from_` feels like the best prefix, but that's kind of taken already on the ensemble itself. Maybe `load_`?

Reply:
"read" is the naming convention used by Pandas and subsequently by Dask for these type of constructor operations. Hence followed that naming convention for tape.

https://pandas.pydata.org/docs/user_guide/io.html
https://docs.dask.org/en/latest/dataframe-api.html#create-dataframes